documentation/docs/25-build-and-deploy/90-adapter-vercel.md

```diff
@@ -153,8 +153,41 @@ If you have Vercel functions contained in the `api` directory at the project's r
 
 Projects created before a certain date may default to using an older Node version than what SvelteKit currently requires. You can [change the Node version in your project settings](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js#node.js-version).
 
-## Troubleshooting
-
 ### Accessing the file system
 
-You can't access the file system through methods like `fs.readFileSync` in Serverless/Edge environments. If you need to access files that way, do that during building the app through [prerendering](https://kit.svelte.dev/docs/page-options#prerender). If you have a blog for example and don't want to manage your content through a CMS, then you need to prerender the content (or prerender the endpoint from which you get it) and redeploy your blog everytime you add new content.
+You can [use files in Serverless Functions on Vercel](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions).
+
+```js
+// @errors: 2307 7031
+/// file: api/pdf/+server.js
+import fs from "node:fs";
+import path from "node:path";
+import { json } from '@sveltejs/kit';
+import PDFDocument from "pdfkit";
+import PalatinoBoldFont from "$lib/fonts/PalatinoBold.ttf";
+
+const font = path.join(process.cwd(), PalatinoBoldFont);
+
+export async function POST({ url }) {
+	const title = url.searchParams.get('title');
+	const filename = url.searchParams.get('filename');
+
+  const doc = new PDFDocument();
+  // use the tmp serverless function folder to create the write stream for the pdf
+  const file = `/tmp/${filename}.pdf`;
+  let writeStream = fs.createWriteStream(file);
+  doc.pipe(writeStream);
+  doc.font(font).fontSize(25).text(title, 100, 100);
```

Comment:
Vite inlines the imported text file so it doesn’t get copied as an asset. I couldn’t think of another example and followed the one provided by Vercel. I’ll try to think up of something else

Reply:
Could you just not import it? My use case was that i'm compiling mjml - I import the templates with ?raw which is great, but the mjml compiler needs to know about an "includes" dir which it grabs files from when it compiles the mjml to html. I spent hours, and hours trying to get this to work on Vercel.

As I mentioned before it might be esoteric, but I'm sure there is another use-case - I've struggled with static files before when trying to grab compiled runtimes out of the "sharp" package and forcing them to deploy to Vercel as it does an fs.readFile at runtime to grab the compiled bundle.

Reply:
I think it has to be imported so Vite’s asset handling kicks in and we grab the processed asset urls off the vite manifest https://vitejs.dev/guide/assets could you share a code example of how your use case is being implemented?

Reply:
Interesting - so I guess my question is, would this work if I wanted to include a directory full of assets, without specifically `import`ing each one. My code looks like this:

```js
// postbuild.js <- run after `vercel build` to deposit files into output dir
import { cp, mkdir } from 'node:fs/promises'
import { join } from 'node:path'

const includesDir = join(
'.',
'.vercel',
'output',
'functions',
'fn.func',
'apps',
'communication',
'includes',
)

const options = { recursive: true }

Promise.all([
  mkdir(includesDir, options),
  cp(
    join('node_modules', '@beyonk', 'email', 'lib', 'includes'),
    includesDir,
    options
  )
])
```

and then in my code I can reference this dir so that the mjml knows where to find its files. No imports required.

```js
const includesDir = join(process.cwd(), 'apps', 'comms')

function compile (template, substitutions) {
  const { html: htmlSource, errors } = toHtml(template.mjml, {
    filePath: includesDir
  })
})
```

Reply:
Yeah, that post-build script should work great https://github.com/sveltejs/kit/issues/10594#issuecomment-1768619317 . This PR is more of convenience, and especially if split functions are enabled.